lib/ruby_lsp/listeners/completion.rb

```diff
@@ -10,16 +10,16 @@ class Completion
       sig do
         params(
           response_builder: ResponseBuilders::CollectionResponseBuilder[Interface::CompletionItem],
-          index: RubyIndexer::Index,
+          global_state: GlobalState,
           nesting: T::Array[String],
           typechecker_enabled: T::Boolean,
           dispatcher: Prism::Dispatcher,
           uri: URI::Generic,
         ).void
       end
-      def initialize(response_builder, index, nesting, typechecker_enabled, dispatcher, uri) # rubocop:disable Metrics/ParameterLists
+      def initialize(response_builder, global_state, nesting, typechecker_enabled, dispatcher, uri) # rubocop:disable Metrics/ParameterLists
         @response_builder = response_builder
-        @index = index
+        @global_state = global_state
```

Comment:
NIT: since we use the index quite a bit, how about we put it in an instance variable for convenience?

```suggestion
        @global_state = global_state
        @index = global_state.index
```